lib/swaps/SwapClient.ts

```diff
@@ -142,43 +143,72 @@ abstract class SwapClient extends EventEmitter {
     });
   }
 
-  protected setStatus = async (status: ClientStatus): Promise<void> => {
+  public init = async () => {
+    // up front checks before initializing client
+    if (!this.isNotInitialized() && !this.isMisconfigured()) {
+      // we only initialize from NotInitialized or Misconfigured status
+      this.logger.warn(`can not init in ${this.status} status`);
+      return;
+    }
+    if (this.disable) {
```

Comment:
Can this be moved before the up front checks?

Reply:
I can, does it make a difference though?

Reply:
Shouldn't make a difference, but from semantically it feels better.